x-pack/plugins/upgrade_assistant/server/lib/es_deprecations_status/get_corrective_actions.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EnrichedDeprecationInfo } from '../../../common/types';
+
+interface Action {
+  action_type: 'remove_settings';
+  objects: string[];
+}
+
+interface Actions {
+  actions: Action[];
+}
+
+type EsMetadata = Actions & {
+  [key: string]: string;
+};
+
+export const getCorrectiveAction = (
+  message: string,
+  metadata: EsMetadata,
+  indexName?: string
+): EnrichedDeprecationInfo['correctiveAction'] => {
+  const indexSettingDeprecation = metadata?.actions?.find(
+    (action) => action.action_type === 'remove_settings' && indexName
+  );
+  const clusterSettingDeprecation = metadata?.actions?.find(
+    (action) => action.action_type === 'remove_settings' && typeof indexName === 'undefined'
+  );
+  const requiresReindexAction = /Index created before/.test(message);
```

Comment:
FWIW I've seen some response messages like:

`Old index with a compatibility version < 8.0`

We should update this logic to surface these as requiring reindex too... I'll ask ES how to reliably get this information from their deprecation API